### PR TITLE
minor filter select enhancements

### PR DESCRIPTION
**Pull request by kkalyan**
* As of now, the max rows for `filter select` is hardcoded to 10K in the code, this PR makes the limit configurable. 
* If there an outstanding ajax request while fetching the filter select values, the previous request is canceled.

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -41,6 +41,8 @@
 
 ROW_LIMIT = 50000
 VIZ_ROW_LIMIT = 10000
+# max rows retrived by filter select auto complete
```
Comment (Mogball) [inline superset/config.py]: retrieved*

File: superset/assets/javascripts/explore/components/controls/FilterControl.jsx [hunk 2]
```diff
@@ -28,6 +28,7 @@ export default class FilterControl extends React.Component {
     }));
     this.state = {
       filters: initialFilters,
+      xhr: null,
```
Comment (Mogball) [inline superset/assets/javascripts/explore/components/controls/FilterControl.jsx]: I would rename this to `activeRequest` or something more descriptive

File: superset/assets/javascripts/explore/components/controls/FilterControl.jsx [hunk 3]
```diff
@@ -45,7 +46,9 @@ export default class FilterControl extends React.Component {
         newStateFilters[index].valuesLoading = true;
         return { filters: newStateFilters };
       });
-      $.ajax({
+      // if there is an outstanding request to fetch values, cancel it.
+      if (this.state.xhr) this.state.xhr.abort();
+      this.setState({ xhr: $.ajax({
```
Comment (Mogball) [inline superset/assets/javascripts/explore/components/controls/FilterControl.jsx]: please format this section of code
```javascript
this.setState({
    xhr: $.ajax({
        ...
    })
});
```

**Review by Mogball — COMMENTED** [review]
Overall looks good. If you could also add 2/3 tests to verify this here https://github.com/apache/incubator-superset/blob/master/superset/assets/spec/javascripts/explore/components/FilterControl_spec.jsx it'd be 👍

File: superset/assets/javascripts/explore/components/controls/FilterControl.jsx (hunk 2 above)
Comment (kkalyan) [reply superset/assets/javascripts/explore/components/controls/FilterControl.jsx]: done.

File: superset/assets/javascripts/explore/components/controls/FilterControl.jsx (hunk 3 above)
Comment (kkalyan) [reply superset/assets/javascripts/explore/components/controls/FilterControl.jsx]: done.

File: superset/config.py (hunk 1 above)
Comment (kkalyan) [reply superset/config.py]: done.

**Review by Mogball — APPROVED** [review]

**Pull request merged**